src/test/engineeffectsdelay_test.cpp

```diff
@@ -0,0 +1,313 @@
+// Tests for engineeffectsdelay.cpp
+
+// Premise: internal Mixxx structure works with a stereo signal.
+// If the mixxx::kEngineChannelCount wouldn't be a stereo in the future,
+// tests have to be updated.
+
+#include "engine/effects/engineeffectsdelay.h"
+
+#include <gtest/gtest.h>
+
+#include <QTest>
+#include <QtDebug>
+
+#include "engine/engine.h"
+#include "test/mixxxtest.h"
+#include "util/sample.h"
+#include "util/types.h"
+
+namespace {
+
+static_assert(mixxx::kEngineChannelCount == mixxx::audio::ChannelCount::stereo(),
+        "EngineEffectsDelayTest requires stereo input signal.");
+
+class EngineEffectsDelayTest : public MixxxTest {
+  protected:
+    void AssertIdenticalBufferEquals(const CSAMPLE* pBuffer,
+            int iBufferLen,
+            CSAMPLE* pReferenceBuffer,
+            int iReferenceBufferLength) {
```

Comment:
Okay, thank you for this tip.

Reply:
I would like to ask you whether you thought `gsl::span` instead of `std::span` from the standard library. I ask this question because Mixxx works with the C++17 standard, but if I'm right, the [span](https://en.cppreference.com/w/cpp/container/span) is in the standard library by C++20.

Reply:
Well it depends. the main branch now does target C++20 (see #4803) (you might need to rebase on the newest commit). For performance-sensitive code, std::span is preferred because it does not do bounds-checking while the safety-focused gsl::span does. 
See https://github.com/microsoft/GSL/wiki/gsl::span-and-std::span

I think if you are still on ubuntu 20.04 you might be stuck with an old gcc version that still lacks adequate c++20 support.

Reply:
Oh, thank you for the explanation. I am using Ubuntu 22.04.